azdo/azdo.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/build"
+)
+
+// ClientFlags is a set of command-line flags always used in go-infra to access the AzDO APIs.
+type ClientFlags struct {
+	// Org is the base URL of the organization, such as 'https://dev.azure.com/dnceng/'
+	Org *string
+	// Proj is the project, such as 'public' or 'internal'.
+	Proj *string
+	// PAT is the Personal Access Token to use.
+	PAT *string
+}
+
+// BindClientFlags creates a ClientFlags struct where each field is set up as a flag.
+func BindClientFlags() *ClientFlags {
+	return &ClientFlags{
+		Org:  flag.String("org", "", "[Required] The AzDO organization URL."),
+		Proj: flag.String("proj", "", "[Required] The AzDO project URL."),
+		PAT:  flag.String("azdopat", "", "[Required] The Azure DevOps PAT to use."),
```

Comment:
I'm missing how these flags end up getting used. It seems this just sets everything to an empty string. Poked around with the `flag` package and can't figure out what I am missing here.

Reply:
Oh so this is parsing the command line params later on?

Reply:
I think looking at the other way to use `flag` first makes it a little clearer--you can pass `flag.StringVar` a pointer to insert the parsed arg value into:

```
var foo string
flag.StringVar(&foo, ...)
flag.Parse()
if foo == "hi" {
...
```

`flag.String` just allocates the memory internally rather than having you pass in your own pointer. You end up with `*string` rather than `string` so `flag.Parse()` can change the value.